src/Schema.php

```diff
@@ -46,6 +49,41 @@ public static function uninstall() {
     Admin::removeAccessCapability();
   }
 
+  /**
+   * Ensures dynamic response headers for all requests from htaccess.
+   */
+  public static function ensureResponseHeadersHtaccess() {
+    $template = file_get_contents(Plugin::getBasePath() . '/conf/.htaccess.headers');
+    if (defined('CORE_STANDARDS_CONTENT_SECURITY_POLICY')) {
+      static::findAndReplaceHeader($template, 'Content-Security-Policy', CORE_STANDARDS_CONTENT_SECURITY_POLICY);
+    }
+    if (defined('CORE_STANDARDS_REFERRER_POLICY')) {
+      static::findAndReplaceHeader($template, 'Referrer-Policy', CORE_STANDARDS_REFERRER_POLICY);
+    }
+    if (defined('CORE_STANDARDS_STRICT_TRANSPORT_SECURITY')) {
+      static::findAndReplaceHeader($template, 'Strict-Transport-Security', CORE_STANDARDS_STRICT_TRANSPORT_SECURITY);
+    }
+    if (defined('CORE_STANDARDS_X_CONTENT_TYPE_OPTIONS')) {
+      static::findAndReplaceHeader($template, 'X-Content-Type-Options', CORE_STANDARDS_X_CONTENT_TYPE_OPTIONS);
+    }
+    if (defined('CORE_STANDARDS_X_XSS_PROTECTION')) {
+      static::findAndReplaceHeader($template, 'X-XSS-Protection', CORE_STANDARDS_X_XSS_PROTECTION);
+    }
+    static::createOrPrependFile(ABSPATH . '.htaccess', $template, "\n");
+  }
+
+  /**
+   * Performs a preg match to replace the new header value in the stream.
+   */
+  private static function findAndReplaceHeader(string &$content, string $headerName, string $headerValue) {
+    $search = sprintf('<(%s) \"(.*)\">', $headerName);
+    // Replace using the same header name and value.
+    $replacement = sprintf('$1 "%s"', $headerValue);
+    $result = preg_replace($search, $replacement, $content);
```

Comment:
Maybe it is just me, but somehow I find the additional separation into `sprintf()` expressions less readable than a direct preg_replace()…?
```suggestion
    $result = preg_replace("@($headerName) \"(.*)\"@", "$1 \"$headerValue\"", $content);
```

Also note that the expression can be simplified: https://regex101.com/r/jQNTde/1 - in particular, the dot does not match a newline unless the modifier "s" is passed.
```suggestion
    $result = preg_replace("@($headerName) .*@", "$1 \"$headerValue\"", $content);
```

Also note that we need to be able to add header expressions/conditions after the value. You can see an example in the headers mentioned in the directives that were referenced in the task's issue summary:
https://github.com/netzstrategen/nutr-nmed/blob/master/.htaccess#L193

This means the quotes should be part of the value that is in the constant:
```suggestion
    $headerName = preg_quote($headerName);
    $result = preg_replace("@($headerName) .*@", "$1 $headerValue", $content);
```
In this final form, the direct preg_replace() definitely looks much easier to read than the separate lines.

Reply:
Using my new approach now there is no need to do any replacement since I'm building the whole template from the variables to reduce complexity.